src/lsm/lsm_cursor.c

```diff
@@ -902,6 +902,107 @@ err:	__clsm_leave(clsm);
 }
 
 /*
+ * __clsm_random_chunk --
+ *	Pick a chunk at random, weighted by the size of all chunks. Weighting
+ * proportional to documents avoids biasing towards small chunks. Then open a
+ * random cursor on the chunk we have picked.
+ */
+static int
+__clsm_random_chunk(WT_SESSION_IMPL *session,
+    WT_CURSOR_LSM *clsm, WT_CURSOR **cursor)
+{
+	WT_DECL_RET;
+	WT_LSM_CHUNK *random_chunk;
+	WT_LSM_TREE *lsm_tree;
+	uint64_t checked_docs, i, rand_doc, total_docs;
+	const char *cfg[3];
+
+	cfg[0] = WT_CONFIG_BASE(session, WT_SESSION_open_cursor);
+	cfg[1] = "next_random=true";
+	cfg[2] = NULL;
+
+	random_chunk = NULL;
+	lsm_tree = clsm->lsm_tree;
+	rand_doc = __wt_random(&session->rnd);
+	__wt_lsm_tree_readlock(session, lsm_tree);
+	for (total_docs = i = 0; i < clsm->nchunks; i++) {
+		total_docs += lsm_tree->chunk[i]->count;
+	}
+
+	rand_doc %= total_docs;
+
+	for (checked_docs = i = 0; i < clsm->nchunks; i++) {
+		checked_docs += lsm_tree->chunk[i]->count;
+		if (rand_doc <= checked_docs) {
+			random_chunk = lsm_tree->chunk[i];
+			break;
+		}
+	}
+	ret = __wt_open_cursor(session, random_chunk->uri, NULL, cfg, cursor);
```

Comment:
Opening a new cursor on every random lookup definitely seems unnecessary.  The `WT_LSM_CURSOR` should already have a cursor at `lsm_cursor->cursors[i]`.  It would be a layering violation to call `__curfile_next_random` on that cursor but we do similar things elsewhere.